### PR TITLE
use cocina factories for specs

### LINKED ISSUE #203: Use cocina factory for spec
To reduce boilerplate.

See https://github.com/sul-dlss/dor-services-app/pull/3948

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made? 🤔

closes #203

## How was this change tested? 🤨

PR is only unit test refactoring (and two tiny opportunistic touchups that don't touch app code)

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test) that use sdr-api*** (e.g.create_object_h2_spec.rb) and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

**Review by jcoyne — CHANGES_REQUESTED** [review]
Can you pull the rubocop changes to a separate PR?  This is a bit challenging to review since it's a substantial change.

File: spec/sdr_client/deposit/process_spec.rb [hunk 1]
```diff
@@ -95,81 +167,7 @@
...
-                      }]
-                    }
-                  }],
-                  hasMemberOrders: [],
-                  isMemberOf: ['druid:gh123df4567']
-                }
-              }.to_json,
-              headers: { 'Content-Type' => 'application/json' }
-            )
+            .with(body: request_dro.to_h.except(:description).to_json, headers: { 'Content-Type' => 'application/json' })
```
Comment (jmartin-sul) [inline spec/sdr_client/deposit/process_spec.rb:170]: i'll admit i traced through the code for a bit, and didn't really understand why i needed to omit `description` here and elsewhere in this spec file.  is this ok, or am i papering over a mistake?

File: spec/sdr_client/deposit/process_spec.rb [hunk 2]
```diff
@@ -255,88 +324,7 @@
...
```
Comment (jmartin-sul) [inline spec/sdr_client/deposit/process_spec.rb:327]: ditto here

**Review by jmartin-sul — COMMENTED** [review]
question on a change i was unsure about

**Comment by jmartin-sul** [comment]
FWIW, there weren't any solely rubocop motivated changes to existing code, just rubocop guidance of my refactoring.  I added the new cops to `.rubocop.yml`, but everything still passed as-is.  And I refreshed `.rubocop_todo.yml because of the `RSpec/MultipleMemoizedHelpers` complaint related to my refactoring of `update_spec.rb`.

So other than the todo file re-gen (needed for this to pass lint), the only rubocop change is `.rubocop.yml`.

Still, happy to split the `.rubocop.yml` change out, or to split into multiple PRs or multiple commits in some other way, if that's helpful.  I can also revisit the `update_spec.rb` changes if the need for the `MultipleMemoizedHelpers` exception should nudge things in a different (less churn-y?) direction.

File: spec/sdr_client/deposit/process_spec.rb (hunk 1 above)
Comment (justinlittman) [reply spec/sdr_client/deposit/process_spec.rb:170]: Looking at the code, it appears that sdr-client does not provide `description` on a request, so I think you're OK.

**Review by justinlittman — CHANGES_REQUESTED** [review]
A few minor comments.

File: spec/sdr_client/deposit/process_spec.rb (hunk 1 above)
Comment (jmartin-sul) [reply spec/sdr_client/deposit/process_spec.rb:170]: thanks!

**Comment by jmartin-sul** [comment]
@justinlittman, thanks for the suggestions, done, i think this is ready for another look

**Review by justinlittman — APPROVED** [review]

**Review by jcoyne — APPROVED** [review]

**Pull request merged**